### DISCUSSION TITLE
AuroKit

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#517] [General Auro discussions]
Something that has been thrown around in different discussions is to provide an omnibus solution to Auro. 

Can we provide a single install version of all the Auro components that are in an approved state of compatibility?

**Comment by blackfalcon** [comment]
Giving this some thought and a few other conversations I want to write down some ideas per this concept. 

Feedback from engineers is that they are overwhelmed and confused with the blasting of updates we do each week. The request of 'why not mono-repo?' isn't really the right question in my mind. I feel it's more rooted in 'how can we get a release cycle that's more consistent?' And that's fair. 

### Proposal

The proposal for this consists of two parts. **First is a new repo called AuroKit.** This repo does not contain any code or host components. It's really only a manifest of components at specified versions guaranteed to interoperately work seamlessly. This could be as simple as a well-defined package.json. The new versioned release of AuroKit would only be released once a quarter.

This would coincide with all the latest releases of all Auro components transitioning from BETA to PROD.

**The second part** of this is curtailing the releases of all the parts of the Auro ecosystem as not to overwhelm users with rapid releases. We can do this via BETA releases. Following this BETA release pattern, ONLY patch releases will be allowed between quarterly releases of AuroKit that affect the prod release SemVer.

**Reply by blackfalcon** [reply]
I REALLY want to move to a BETA release model to allow us to rapidly release updates to components, but with the clear understanding that this is a new BETA version and there are risks for early adopters. If we had done this with the changes to Tokens and WCSS, this could have really saved everyone a lot of frustration.

**Reply by braven112** [reply]
Looks like the core issue to solve is that components and their dependencies need to be working together when they are released.  I like the Beta release idea to go with this so that teams can still move faster if necessary.  

In practice, quarterly-ish might work but ensuring that dependencies work together when we release cross component features should dictate when we release and that could happen sooner or later than a set schedule.

**Comment by blackfalcon** [comment]
Notes from a side chat with Lindsey. These are questions that need to be answered before work begins on this concept. 

1. Will we re-release the whole auro kit for every one-off change?  
2. Are we doing it once or twice a year and the rest will be component only?  
3. What problem are we solving and does this introduce more?

**Comment by blackfalcon** [comment]
Link to POC repo: 

- https://github.com/AlaskaAirlines/auro-kit
